powertools-parameters/src/main/java/software/amazon/lambda/powertools/parameters/internal/LambdaParametersAspect.java

```diff
@@ -0,0 +1,42 @@
+package software.amazon.lambda.powertools.parameters.internal;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.aspectj.lang.reflect.FieldSignature;
+import software.amazon.lambda.powertools.parameters.*;
+
+@Aspect
+public class LambdaParametersAspect {
+
+    @Pointcut("get(* *) && @annotation(paramAnnotation)")
+    public void getParam(Param paramAnnotation) {
+    }
+
+    @Around("getParam(paramAnnotation)")
+    public Object injectParam(final ProceedingJoinPoint joinPoint, final Param paramAnnotation) {
+        BaseProvider provider = ParamManager.getProvider(paramAnnotation.provider());
+        if(null == provider) {
+            throw new IllegalArgumentException(String.format("ParamProvider %s not supported.", paramAnnotation.provider().getName()));
```

Comment:
When will this exception occur ? only when clients set provider as null explicitly rite ? In that case I think we should improve the message that provider cannot be null ? or is there some other scenario where this can be null ? 🤔

Reply:
indeed only when client set provider as null. Moving the null check as first block on the method and changing the message to "provider cannot be null"